uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/annual-entitlements.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/absence-tab/modules/components',
+  'common/models/contact'
+], function (_, moment, components) {
+  components.component('annualEntitlements', {
+    bindings: {
+      contactId: '<'
+    },
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/annual-entitlements.html';
+    }],
+    controllerAs: 'entitlements',
+    controller: ['$log', '$q', 'AbsenceType', 'AbsencePeriod', 'Entitlement', 'Contact', controller]
+  });
+
+  function controller ($log, $q, AbsenceType, AbsencePeriod, Entitlement, Contact) {
+    $log.debug('Component: annual-entitlements');
+
+    var vm = {};
+
+    vm.absenceTypes = [];
+    vm.absencePeriods = [];
+    vm.contactId = this.contactId;
+    vm.contacts = [];
+    vm.entitlements = [];
+    vm.loaded = { absencePeriods: false };
+
+    (function init () {
+      return $q.all([
+        loadAbsenceTypes(),
+        loadEntitlements()
+      ])
+      .then(function () {
+        return loadCommentsAuthors();
+      })
+      .then(function () {
+        return loadAbsencePeriods();
+      });
```

Comment:
Loader should be hidden in `finally`which would work if there is an error also, also readability is better
```javascript
.finally(function() {
  vm.loaded.absencePeriods = true;
})
```
Same should be done for `contractEntitlements `